src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java

```diff
@@ -179,58 +148,69 @@ public CreateCaseEventResult createCaseEvent(String caseReference, CaseDataConte
-        boolean isApiVersion3 = request.getContentType() != null
-            && request.getContentType().equals(V3.MediaType.CREATE_EVENT);
-
-        CaseDocumentAttacher caseDocumentAttacher = null;
-        if (isApiVersion3) {
-            validateCaseFieldsOperation.validateData(content.getData(), caseTypeDefinition, content);
-            caseDocumentAttacher = new CaseDocumentAttacher(restTemplate, applicationParams, securityUtils);
-            caseDocumentAttacher.extractDocumentsWithHashTokenBeforeCallbackForUpdate(content.getData(),
-                caseDetailsBefore);
-        }
 
-        mergeUpdatedFieldsToCaseDetails(content.getData(), caseDetails, caseEventDefinition, caseTypeDefinition);
-        AboutToSubmitCallbackResponse aboutToSubmitCallbackResponse =
-            callbackInvoker.invokeAboutToSubmitCallback(caseEventDefinition,
-                caseDetailsBefore,
-                caseDetails,
-                caseTypeDefinition,
-                content.getIgnoreWarning());
+        final CaseDetails updatedCaseDetails = mergeUpdatedFieldsToCaseDetails(
+            content.getData(),
+            caseDetails,
+            caseEventDefinition,
+            caseTypeDefinition
+        );
+        final CaseDetails updatedCaseDetailsWithoutHashes = caseDocumentService.stripDocumentHashes(updatedCaseDetails);
+
+        final AboutToSubmitCallbackResponse aboutToSubmitCallbackResponse = callbackInvoker.invokeAboutToSubmitCallback(
+            caseEventDefinition,
+            caseDetailsInDatabase,
+            updatedCaseDetailsWithoutHashes,
+            caseTypeDefinition,
+            content.getIgnoreWarning()
+        );
 
         final Optional<String> newState = aboutToSubmitCallbackResponse.getState();
 
-        validateCaseFieldsOperation.validateData(caseDetails.getData(), caseTypeDefinition, content);
-        LocalDateTime timeNow = now();
+        @SuppressWarnings("UnnecessaryLocalVariable")
+        final CaseDetails caseDetailsAfterCallback = updatedCaseDetailsWithoutHashes;
 
-        if (isApiVersion3) {
-            caseDocumentAttacher.caseDocumentAttachOperation(caseDetails, newState.isPresent());
-            caseDocumentAttacher.findDifferenceWithExistingCaseDetail(caseDetailsBefore,caseDetails);
-        }
+        validateCaseFieldsOperation.validateData(caseDetailsAfterCallback.getData(), caseTypeDefinition, content);
+        final LocalDateTime timeNow = now();
 
-        String oldState = caseDetails.getState();
-        final CaseDetails savedCaseDetails =
-            saveCaseDetails(caseDetailsBefore,
-                caseDetails,
-                caseEventDefinition,
-                newState,
-                timeNow);
-        saveAuditEventForCaseDetails(aboutToSubmitCallbackResponse,
+        final List<DocumentHashToken> documentHashes = caseDocumentService.extractDocumentHashToken(
+            Optional.ofNullable(caseDetailsInDatabase.getData()).orElse(emptyMap()),
```

Comment:
in the database `data` column is not null. so safe null checks may not be needed.